test/localstack.ts

```diff
@@ -38,21 +42,29 @@ export const deployStack = async (app: App, stack: Stack) => {
     credentials: clientConfig.credentials,
   };
 
+  performance.mark("CfnCompile");
+
   const cfn = new CloudFormationDeployments({
     sdkProvider,
   });
 
-  const assembly = new cxapi.CloudAssembly(cloudAssembly.directory);
-  const stackArtifact = cxapi.CloudFormationStackArtifact.fromManifest(
-    assembly,
-    stack.artifactId,
-    cloudAssembly.getStackArtifact(stack.artifactId).manifest
-  ) as cxapi.CloudFormationStackArtifact;
+  // const assembly = new cxapi.CloudAssembly(cloudAssembly.directory);
+  // const stackArtifact = cxapi.CloudFormationStackArtifact.fromManifest(
+  //   cloudAssembly,
+  //   stack.artifactId,
+  //   cloudAssembly.getStackArtifact(stack.artifactId).manifest
+  // ) as cxapi.CloudFormationStackArtifact;
+
+  performance.measure("Complete CfnCompile", "CfnCompile");
 
+  performance.mark("DeployStack");
   await cfn.deployStack({
-    stack: stackArtifact,
+    stack: cloudAssembly.getStackArtifact(
+      stack.artifactId
+    ) as unknown as cxapi.CloudFormationStackArtifact,
```

Comment:
How does this work? What is `cloudAssembly`?

Reply:
Cloud assembly is what CDK calls the output of a stack synth.

Reply:
Original inspiration: https://github.com/aws/aws-cdk/pull/18667#issuecomment-1075348390